devito/arch/archinfo.py

```diff
@@ -302,6 +303,38 @@ def homogenise_gpus(gpu_infos):
     return None
 
 
+@memoized_func
+def detect_cc():
+    CUDA_SUCCESS = 0
+
+    libnames = ('libcuda.so', 'libcuda.dylib', 'cuda.dll')
+    for libname in libnames:
+        try:
+            cuda = ctypes.CDLL(libname)
+        except OSError:
+            continue
+        else:
+            break
+    else:
+        # warning("could not load any of: " + ' '.join(libnames))
+        return None
+
+    cc_major = ctypes.c_int()
+    cc_minor = ctypes.c_int()
+    result = ctypes.c_int()
+    device = ctypes.c_int()
+    error_str = ctypes.c_char_p()
+
+    result = cuda.cuInit(0)
+    if result != CUDA_SUCCESS:
+        cuda.cuGetErrorString(result, ctypes.byref(error_str))
+        # print("cuInit failed with error code %d: %s" % (result, error_str.value.decode()))
+        return None
+    if (cuda.cuDeviceComputeCapability(ctypes.byref(cc_major),
+       ctypes.byref(cc_minor), device) == CUDA_SUCCESS):
```

Comment:
this is not how we indent code in devito, would be surprised if flake8 wouldn't throw an error

Reply:
Yes, my bad